### PR TITLE
feat: :sparkles: Add EPIC's IdToken parsing and verification

### PR DESCRIPTION
**Pull request by lucasthomazoni**

### CODE REVIEW

File: lib/smart_on_fhir/epic.go [hunk 1]
```diff
@@ -37,39 +35,37 @@ func NewEpicSmartOnFhirProvider(credentials *Credentials) *FhirProvider {
...
+					hasJwks = true
+					break
+				}
 			}
-			defer resp.Body.Close()
 
-			body, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
+			if hasJwks {
+				fmt.Println("=========== Key ID matches ===========")
```
Comment (xcskier56) [inline lib/smart_on_fhir/epic.go:74]: Are you planning on verifying the ID token using the key that we found in this PR or next?

Comment (lucasthomazoni) [reply lib/smart_on_fhir/epic.go:74]: Which kind of verification do you want here? Isn't it just comparing the `kid`'s here?

Comment (xcskier56) [reply lib/smart_on_fhir/epic.go:74]: We need to use the jwk to do a verifying decode of the `id_token`. Here's some psuedo ruby code:

```ruby
id_token = Jwt.decode_no_verify(res.id_token)
keys = fetch_jwks
key = nil

# Find and set the key
keys.each do |k|
  if k['kid'] == id_token['kid']
    key = k
    break
  end
end

# Likely: Convert JWK into PEM or similar
pem = convert(jwk)

# Use the KID to verify the signature of the ID token
token, err = validate(id_token, pem)

raise 'token not validated' if err

# ...
```

Comment (xcskier56) [reply lib/smart_on_fhir/epic.go:74]: Converting the JWK into a format that the validate function can use is the hard part. A PEM might work but I'm not 100% sure exactly what format the Go JWT library needs

**Comment by lucasthomazoni** [comment]
@xcskier56 Added token signature verification, have I told you how much I hate golang? jeez

**Review by xcskier56 — APPROVED** [review]
Looks good to me! Thanks for sticking through all the little issues and getting this across the finish line! 🥳

**Pull request merged**